source/core/src/main/com/csse3200/game/components/minigame/snake/SnakeGrid.java

```diff
@@ -0,0 +1,45 @@
+package com.csse3200.game.components.minigame.snake;
+
+import com.badlogic.gdx.Gdx;
+import com.badlogic.gdx.math.Vector2;
+import com.csse3200.game.components.minigame.Grid;
+
+/**
+ * Represents a specialised grid for the Snake game.
+ * The SnakeGrid has default dimensions and includes methods for snake-specific functionality.
+ */
+public class SnakeGrid extends Grid {
+
+    private static final int DEFAULT_WIDTH = 20;
+    private static final int DEFAULT_HEIGHT = 20;
+
+    /**
+     * Creates a SnakeGrid with default dimensions.
+     * The grid is initialized to a size of 20x20 cells.
+     */
+    public SnakeGrid() {
+        super(DEFAULT_WIDTH, DEFAULT_HEIGHT);
+    }
+
+    public boolean isCollision(int x, int y) {
```

Comment:
Are these methods used/will they be used? If they will be used they need unit tests (should be pretty simple for tileToWorldPosition at least, just assert that the mathematical operations work as expected)